packages/machine/src/action.ts

```diff
@@ -104,15 +89,21 @@ export class ActionExecution {
         context
       );
       this.instructionPointer += 1;
-      this.results.push({ value, opCode: internalMessage.opCode });
+
+      // push modified value of `context.intermediateResults`
+      this.intermediateResults = context.intermediateResults;
 
       return { value, done: false };
     } catch (e) {
-      throw Error(
-        `While executing op ${Opcode[internalMessage.opCode]} at seq ${
-          this.clientMessage.seq
-        }, execution failed with the following error. ${e.stack}`
-      );
+      if (e.stack.search("Crashing the machine on purpose") === -1) {
```

Comment:
@IIIIllllIIIIllllIIIIllllIIIIllllIIIIll We need to figure out another way to handle this. I feel test-related circumstances should be resolved in their context, not in the real codebase. Happy to help out with this 🤚

Reply:
will revert